flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/AbstractStreamOperator.java

```diff
@@ -887,6 +864,30 @@ public void close() {
 	//  Watermark handling
 	// ------------------------------------------------------------------------
 
+	public <K> void registerWatermarkCallback(KeyRegistry.OnWatermarkCallback<K> callback, TypeSerializer<K> keySerializer) {
```

Comment:
The key serialiser can be retrieved from `getKeyedStateBackend().getKeySerializer()`.